extensions/dicom-rt/src/components/RTPanel/RTPanel.js

```diff
@@ -0,0 +1,211 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import cornerstoneTools from 'cornerstone-tools';
+import cornerstone from 'cornerstone-core';
+
+import { ScrollableArea, TableList, Icon } from '@ohif/ui';
+
+import './RTPanel.css';
+import StructureSetItem from '../StructureSetItem/StructureSetItem';
+import RTPanelSettings from '../RTSettings/RTSettings';
+import PanelSection from '../PanelSection/PanelSection';
+
+const refreshViewport = () => {
+  cornerstone.getEnabledElements().forEach(enabledElement => {
+    cornerstone.updateImage(enabledElement.element);
+  });
+};
+
+/**
+ * RTPanel component
+ *
+ * @param {Object} props
+ * @param {Array} props.studies
+ * @param {Array} props.viewports - viewportSpecificData
+ * @param {number} props.activeIndex - activeViewportIndex
+ * @param {number} props.isOpen - isOpen
+ * @returns component
+ */
+const RTPanel = ({ studies, viewports, activeIndex, isOpen, onContourItemClick }) => {
+  const [selectedContour, setSelectedContour] = useState();
+  const DEFAULT_SET_INDEX = 0;
+  const DEFAULT_STATE = {
+    sets: [],
+    contours: [],
+    selectedSet: null,
+  };
+
+  const [state, setState] = useState(DEFAULT_STATE);
+  const [showSettings, setShowSettings] = useState(false);
+
+  useEffect(() => {
+    document.addEventListener('rtloaded', updateStructureSets);
+
+    return () => {
```

Comment:
@dannyrb Ill need your help figuring out a way to handle this. The user can open the panel while the rtstruct sets are being loaded, if it happens, the component its not gonna react to the changes, considering the current implementation of extensions / rtload logic, any ideas?

Reply:
In an ideal world, subscribe to `MeasurementService` in a `useEffect` hook.
Then update local state when subscriptions are fired?

For now, even though this is hacky, I think its your only option given time constraints/budget.

My recommendation: Document that this is not ideal (and articulate why), and how you would recommend improving it. Spend less than 45 minutes, but do try to capture the essence of the problem and the value of resolving it so that a future implementer isn't lost, and it's easier to prioritize in our backlog.

@JamesAPetts may weigh in differently. Just my 2 cents.

Reply:
For the sake of namespacing and adding event listeners to `document`, should probably call the event `extensiondicomrtrtloaded`, so its unique enough.

Reply:
Updated. thanks.